File: 20260618/exos/02_condition.py
# ============================================================
# Project Zomboid — niveau de faim
# ============================================================
# L'HUD affiche un mot selon la faim du survivant :
#   plus de 80  →  "Rassasié"
#   plus de 30  →  "Correct"
#   sinon       →  "Affamé"
#
# Ton personnage a faim = 65. Quel mot s'affiche ?
#
# Lance : python3 02_condition.py
# ============================================================

faim = 65
statut = ""

# Détermine statut avec if / elif / else.
#
# faim = 65
#   65 > 80 ?  non
#   65 > 30 ?  oui  →  statut = "Correct"
#
# (si faim valait 90  →  "Rassasié" ; si 10  →  "Affamé")
#
# Résultat attendu : statut == "Correct"
#
# Indice : if / elif / else

# À toi :
if faim > 80:
    statut = "Rassasié"
elif faim <= 80 and faim > 30:
    statut = "Correct"
else:
    statut ="Affamé"

# --- Vérification (ne pas modifier) ---
assert statut == "Correct", f"Obtenu : '{statut}', attendu : 'Correct'"
print("✅ Correct !")

File: 20260721/exos/02_condition.lua
-- ============================================================
-- Project Zomboid — alerte faim
-- ============================================================
-- Si la faim est à 70 ou plus, le jeu affiche "Affame".
-- Sinon, "Ok".
--
-- Lance : lua 02_condition.lua
-- ============================================================

faim = 75
alerte = ""

-- Remplis alerte selon la valeur de faim.
--
-- faim = 75  →  75 >= 70 ? oui  →  alerte = "Affame"
-- faim = 40  →  40 >= 70 ? non  →  alerte = "Ok"
--
-- Résultat attendu : alerte == "Affame"  (car faim vaut 75 ici)
--
-- Indice : if / else / end

-- À toi :
if faim >= 70 then
    alerte = "Affame"
else
    alerte = "Ok"
end

-- --- Vérification (ne pas modifier) ---
assert(alerte == "Affame", "Obtenu : '" .. alerte .. "' — avec faim = 75, attendu : Affame")
print("✅ Correct !")
